build_tools/update_environments_and_lock_files_pr.py

```diff
@@ -0,0 +1,47 @@
+import argparse
+import os
+import shlex
+import subprocess
+
+
+def execute_command(command):
+    command_list = shlex.split(command)
+    subprocess.run(command_list, check=True, text=True)
+
+
+def main():
+    comment = os.environ["COMMENT"].splitlines()[0].strip()
+
+    # Extract the command-line arguments from the comment
+    prefix = "@scikit-learn-bot update lock-files"
+    assert comment.startswith(prefix)
+    args_string = comment[len(prefix) :]  # including any leading spaces
+
+    # Extract the tag from the args (if any)
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--select-tag", default=None)
+    args, _ = parser.parse_known_args(shlex.split(args_string))
+    tag = args.select_tag
+
+    # Determine the marker to add in the commit message (if any)
+    if tag == "main-ci":
+        marker = "[doc build] "
+    elif tag == "scipy-dev":
+        marker = "[scipy-dev] "
+    elif tag == "arm":
+        marker = "[cirrus arm] "
+    else:
+        marker = ""
```

Comment:
I think by default this updates all the lock-files so all the markers should be present `[doc build] [scipy-dev] [cirrus arm]`. Not sure this will happen that often in practice though that you want to update all the lock-files at once ...